security/fscrypt-tests.py

```diff
@@ -0,0 +1,71 @@
+#!/usr/bin/env python
+#
+# This program is free software; you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation; either version 2 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
+#
+# See LICENSE for more details.
+#
+# Copyright: 2021 IBM
+# Author: Nageswara R Sastry <rnsastry@linux.ibm.com>
+
+import os
+from avocado import Test
+from avocado.utils import archive, build, distro, linux_modules
+from avocado.utils.software_manager import SoftwareManager
+
+
+class fscrypt(Test):
+
+    """
+    fscrypt-testsuite
+    :avocado: tags=security,testsuite
+    """
+
+    def setUp(self):
+        '''
+        Install the basic packages to support fscrypt
+        '''
+        cfg_param = "CONFIG_FS_ENCRYPTION"
+        ret = linux_modules.check_kernel_config(cfg_param)
+        if ret == linux_modules.ModuleConfig.NOT_SET:
+            self.cancel("%s not set." % cfg_param)
+        else:
+            self.log.info("%s set." % cfg_param)
+        # Check for basic utilities
+        smm = SoftwareManager()
+        detected_distro = distro.detect()
+        deps = ['gcc', 'make']
+        if detected_distro.name in ['rhel', 'SuSE', 'fedora', 'centos']:
+            deps.extend(["autoconf", "automake", "golang", "expect",
+                         "keyutils", "e2fsprogs", "pam-devel"])
+        else:
+            self.cancel("Unsupported distro %s for fscrypt package"
+                        % detected_distro.name)
+        for package in deps:
+            if not smm.check_installed(package) and not smm.install(package):
+                self.cancel('%s is needed for the test to be run' % package)
+        url = "https://github.com/google/fscrypt/archive/refs/heads/master.zip"
```

Comment:
ah it tricky , as above source has Apache-2.0 License  and in misc test repo we have GPU  License  not sure we can call this from this repo @narasimhan-v can you please suggest , 

@nasastry did not started review , let us first have more clarity on  above query than we can have review and merge

Reply:
@PraveenPenguin @nasastry any further discussion on this?

Reply:
@PraveenPenguin we are fine with GPL and Apache License, both are almost similar open source free software liscence